internal/services/mssql/mssql_database_resource.go

```diff
@@ -87,6 +87,18 @@ func resourceMsSqlDatabase() *pluginsdk.Resource {
 					}
 				}
 
+				if strings.HasPrefix(skuName, "DW") {
+					// NOTE: Got `PerDatabaseCMKDWNotSupported` error from API when `sku_name` is set to `DW100c` and `transparent_data_encryption_key_vault_key_id` is specified
+					keyVaultKeyId := d.Get("transparent_data_encryption_key_vault_key_id").(string)
+					if keyVaultKeyId != "" {
+						return fmt.Errorf("database-level CMK is not supported for Data Warehouse SKUs")
+					}
+					// NOTE: Got `InternalServerError` error from API when `sku_name` is set to `DW100c` and `transparent_data_encryption_key_automatic_rotation_enabled` is specified
+					if _, ok := d.GetOk("transparent_data_encryption_key_automatic_rotation_enabled"); ok {
```

Comment:
This seems opposite to your PR comment that this property cannot be set to false (the default) when using a DW SKU?

Reply:
The PR comment has been updated below. Please feel free let me know if you have any feedback, thank you.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/f17c3534-2ee1-4673-91c4-e621f15c0860)

Reply:
Thanks, but I still don't think I understand. If encryptProtectorAutoRotation must be true for DW SKUs, then the user must specify `transparent_data_encryption_key_automatic_rotation_enabled = true`, but your change prevents them from specifying it?

Shouldn't this be:

```suggestion
					if !d.Get("transparent_data_encryption_key_automatic_rotation_enabled").(bool) {
						return fmt.Errorf("`transparent_data_encryption_key_automatic_rotation_enabled` must be `true` when using a Data Warehouse SKU")
```

Reply:
Yes, what is special about the API behavior is that this field must be absent (neither true nor false) for DW SKUs. Otherwise API returns `InternalServerError`.

Reply:
@sinbai I think in this case, we should test for `true` here and raise an error accordingly, but allow a `false` value. In the Create and Update functions, we can choose to omit the `EncryptionProtectorAutoRotation` field (set to `nil`) if `transparent_data_encryption_key_automatic_rotation_enabled = false` and `sku ~= "dw..."` - this will avoid breaking any older configurations and/or state that might have previously set this.

Conversely, we can set the property to `false` in the Read function when using a DW SKU and the field is absent in the response.

```suggestion
				if d.Get("transparent_data_encryption_key_automatic_rotation_enabled").(bool) {
```